kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/DataModelUtil.java

```diff
@@ -207,4 +207,38 @@ public static Schema makeNullableSchema(Schema schema) {
     return new EntityAccessor<E>(type, schema);
   }
 
+  public static <E> E deepCopy(E entity, Schema dstSchema) {
```

Comment:
These may be good to have later, but if they aren't needed for this commit let's remove them. We can add them as a separate feature and add appropriate tests. Also, if what we want is a schema update then you might want to do that instead of a deep copy. There's no need to deep copy when the schema for a sub-record hasn't changed.